spring-boot-autoconfigure-data-aerospike/src/test/java/org/springframework/boot/autoconfigure/data/aerospike/AerospikeDataAutoConfigurationTests.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.data.aerospike;
+
+import com.aerospike.client.AerospikeClient;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.data.aerospike.core.AerospikeTemplate;
+import org.springframework.data.aerospike.core.ReactiveAerospikeTemplate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link AerospikeDataAutoConfiguration}.
+ *
+ * @author Igor Ermolenko
+ */
+public class AerospikeDataAutoConfigurationTests {
+    private ApplicationContextRunner contextRunner = new ApplicationContextRunner()
```

Comment:
Is it possible to test content without specific classes being present on the classpath? Would be very useful.
I've seen this: https://github.com/spring-cloud/spring-cloud-commons/tree/master/spring-cloud-test-support/src/main/java/org/springframework/cloud/test . Could you please check whether these utils are already present in spring-boot-starter official release?

Reply:
No, **spring-boot-starter** does not include **spring-cloud-test-support**.
But we can add it as regular dependency:
`
<dependency>
    <groupId>org.springframework.cloud</groupId>
    <artifactId>spring-cloud-test-support</artifactId>
    <version>2.2.0.RELEASE</version>
    <scope>test</scope>
</dependency>
`

Reply:
@igor-ermolenko https://spring.io/blog/2018/03/07/testing-auto-configurations-with-spring-boot-2-0
```
@Test
public void serviceIsIgnoredIfLibraryIsNotPresent() {
    this.contextRunner
        .withClassLoader(new FilteredClassLoader(UserService.class))
        .run((context) -> assertThat(context)
                .doesNotHaveBean("userService"));
}
```
can you check? this should probably suit us better than `spring-cloud-test-support`